### PR TITLE
Ignore JWT expiration when refreshing the Token: Refresh limit should…

### PR DESCRIPTION
**Pull request by tleguijt**
See discussion #92.
This is a proposal for a fix

### CODE REVIEW

File: tests/test_views.py [hunk 1]
```diff
@@ -356,6 +356,23 @@ def test_refresh_jwt_after_refresh_expiration(self):
         self.assertEqual(response.data['non_field_errors'][0],
                          'Refresh has expired.')
 
+    def test_refresh_jwt_after_jwt_expiration(self):
+        """
+        Test that token can be refreshed after token expiratoin but before token refresh limit
```
Comment (Gaffney) [inline tests/test_views.py]: typo : `expiratoin` -> `expiration`

**Comment by jpadilla** [comment]
I'm still not sure about this. I keep thinking there's a possible security issue allowing this. @tleguijt where was the original discussion, can't seem to find it.

**Comment by oneTimePad** [comment]
+1 about security issue. If the user attempts to refresh with an expired token that should result in a forced re-login (invalid refresh); they should be forced to resend their credentials to obtain an new token. Else the whole idea of refresh becomes pointless.

**Comment by tleguijt** [comment]
@jpadilla I've linked the discussion in my first comment.

I share your doubts about security, still, I tend to think this should be the intended implementation;

> A refresh token allows the application to request Auth0 to issue a new id_token directly, without needing to reauthenticate. This works as long as the refresh token has not been revoked.
from https://auth0.com/docs/refresh-token

and

> IMPORTANT ADVICE: obtaining new tokens using the refresh_token should happen only if the id_token has expired. For example, it is a bad practice to call the endpoint to get a new token every time you do an API call. There are rate limits in Auth0 that will throttle the amount of requests that can be done using the same token from a certain IP to this endpoint.

**Comment by oneTimePad** [comment]
That documentation is talking about long-time refresh tokens, a special extra token. To make it more secure a long time refresh token should be generated to allow a user to refresh and fetch a new short-time token after it has expired but checks the expiration of the long time refresh token instead

**Comment by ticosax** [comment]
I'm jumping to advertise the existence of such application. https://github.com/lock8/django-rest-framework-jwt-refresh-token

**Pull request closed without merging**